htdocs/multicurrency/multicurrency_rate.php

```diff
@@ -229,163 +236,124 @@
-		print '<input type="hidden" name="token" value="'.newToken().'">';
-
-		print '<table><tr>';
-		print ' <td>'.$langs->trans('Date').'</td>';
-		print '<td>';
-		print $form->selectDate($current_rate->date_sync, 'dateinput');
-		print '</td>';
+$sql = 'SELECT cr.rowid, cr.date_sync, cr.rate, cr.entity, m.code, m.name ';
 
-		print '<td> '.$langs->trans('Currency').'</td>';
-		print '<td>'.$form->selectMultiCurrency($currency_code, 'multicurrency_code', 0, " code != '".$conf->currency."'", true).'</td>';
+// Add fields from hooks
+$parameters=array();
+$reshook=$hookmanager->executeHooks('printFieldListSelect', $parameters);    // Note that $action and $object may have been modified by hook
+$sql .= $hookmanager->resPrint;
+$sql .= ' FROM '.MAIN_DB_PREFIX.'multicurrency_rate as cr ';
+$sql .=" INNER JOIN ".MAIN_DB_PREFIX."multicurrency AS m ON cr.fk_multicurrency = m.rowid";
 
-		print '<td>'.$langs->trans('Rate').'</td>';
-		print '<td><input class="minwidth200" name="rateinput" value="'.dol_escape_htmltag($current_rate->rate).'" type="text"></td>';
 
-		print '<td>';
-		print '<input type="hidden" name="action" value="update">';
-		print '<input type="hidden" name="id_rate" value="'.$current_rate->id.'">';
-		print '<input type="submit" class="butAction" name="btnupdateCurrencyRate" value="'.$langs->trans('UpdateRate').'">';
-		print '<a href="'.$_SERVER["PHP_SELF"].'" class="butAction">'.$langs->trans('CancelUpdate').'</a>';
+if ($sall) $sql .= natural_search(array_keys($fieldstosearchall), $sall);
 
-		print '</td>';
-		print '</tr></table>';
-		print '</form>';
-	} else {
-		dol_syslog("currency_rate:list:update", LOG_WARNING);
-	}
+if ($search_date_sync && $search_date_sync_end ){
+    $sql .= " AND (cr.date_sync BETWEEN '".dol_escape_htmltag($search_date_sync)."' AND '".dol_escape_htmltag($search_date_sync_end)."')";
+} elseif ($search_date_sync && !$search_date_sync_end) {
+    $sql .= natural_search('cr.date_sync', $search_date_sync);
 }
 
 
-$sql = 'SELECT cr.rowid, cr.date_sync, cr.rate, cr.entity, m.code, m.name ';
-// Add fields from hooks
-$parameters = array();
-$reshook = $hookmanager->executeHooks('printFieldListSelect', $parameters); // Note that $action and $object may have been modified by hook
-$sql .= $hookmanager->resPrint;
-$sql .= ' FROM '.MAIN_DB_PREFIX.'multicurrency_rate as cr ';
-$sql .= " INNER JOIN ".MAIN_DB_PREFIX."multicurrency AS m ON cr.fk_multicurrency = m.rowid";
-if ($sall) $sql .= natural_search(array_keys($fieldstosearchall), $sall);
-if ($search_date_sync) $sql .= natural_search('cr.date_sync', $search_date_sync);
-if ($search_rate) $sql .= natural_search('cr.rate', $search_rate);
-if ($search_code) $sql .= natural_search('m.code', $search_code);
-$sql .= " WHERE m.code <> '".$db->escape($conf->currency)."'";
+if ($search_rate)   $sql .= natural_search('cr.rate', $search_rate);
+if ($multicurrency_code) $sql .= natural_search('m.code', $multicurrency_code);
```

Comment:
Why renaming the filter into another variable name.
All filters of a list now follow rule search_xxxxx (this is used by internal features)

Reply:
agree with that. modified.